### PR TITLE
devices: MIMX8ML8: Add support for new IRQSTEER driver

### PR DESCRIPTION
**Pull request by LaurentiuM1234**
Depends on https://github.com/nxp-mcuxpresso/mcux-sdk/pull/146.

### CODE REVIEW

**Comment by LaurentiuM1234** [comment]
Update: MCUX-SDK changes have been merged

**Comment by LaurentiuM1234** [comment]
New MCUX-SDK dependency: https://github.com/nxp-mcuxpresso/mcux-sdk/pull/151

File: mcux/mcux-sdk/drivers/irqsteer/fsl_irqsteer.c [hunk 1]
```diff
@@ -144,7 +201,41 @@ IRQn_Type IRQSTEER_GetMasterNextInterrupt(IRQSTEER_Type *base, irqsteer_int_mast
...
+            }
+
+            bitOffset++;
+            chanStatus = chanStatus >> 1U;
+        }
+    }
+
+    return NotAvail_IRQn;
+}
+#endif /* __CORTEX_M */
```
Comment (iuliana-prodan) [inline mcux/mcux-sdk/drivers/irqsteer/fsl_irqsteer.c:237]: Is missing IRQSTEER_GetMasterInterruptsStatus() which is here: https://github.com/nxp-mcuxpresso/mcux-sdk/pull/151/files#diff-4e14c6250d0b93ae23138a5afbfff872604276eebc0e3678a388871ee50f6200R238.

Comment (LaurentiuM1234) [reply mcux/mcux-sdk/drivers/irqsteer/fsl_irqsteer.c:237]: Weird, I can see it being added just below your comment:

screen

Am I missing something here?

Comment (iuliana-prodan) [reply mcux/mcux-sdk/drivers/irqsteer/fsl_irqsteer.c:237]: This is on my end:
image

Comment (iuliana-prodan) [reply mcux/mcux-sdk/drivers/irqsteer/fsl_irqsteer.c:237]: Sorry, my bad... I was on a wrong commit, not all changes.

**Review by iuliana-prodan — APPROVED** [review]
Good to go! 👍

File: mcux/mcux-sdk/drivers/irqsteer/fsl_irqsteer.c (hunk 1 above)
Comment (LaurentiuM1234) [reply mcux/mcux-sdk/drivers/irqsteer/fsl_irqsteer.c:237]: Ah, ok. Was very confused about what was happening here. Would've been a pretty funny issue though

**Comment by LaurentiuM1234** [comment]
@mmahadevan108 @dleach02 I think this is good to go. All MCUX-SDK dependencies have been merged. Zephyr PR should also be mergeable at this point.

**Pull request merged**